src/llm/client.py

```
@@ -1,48 +1,128 @@
 import logging
 from typing import Dict, List, Any, Optional
 from ..config.logging_config import get_logger
 from ..config.model_config import get_model_config, get_generation_config
+from ..utils.errors import classify_and_log_genai_error
+
 
 logger = get_logger(__name__)
+# Optional: SDK-specific errors and HTTP client timeout classes
+try:
+    from google.genai import errors as genai_errors  # type: ignore
+except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)